polkadot/xcm/xcm-builder/src/location_conversion.rs

```diff
@@ -84,6 +84,20 @@ impl DescribeLocation for DescribeAccountKey20Terminal {
 	}
 }
 
+/// Create a description of the remote treasury `location` if possible. No two locations should have
+/// the same descriptor.
+pub struct DescribeTreasuryVoiceTerminal;
+
+impl DescribeLocation for DescribeTreasuryVoiceTerminal {
+	fn describe_location(l: &MultiLocation) -> Option<Vec<u8>> {
+		match (l.parents, &l.interior) {
+			(0, X1(Plurality { id: BodyId::Treasury, part: BodyPart::Voice })) =>
+				Some(("Treasury").encode()),
```

Comment:
I am not sure, is this enough?
If I check `DescribeBodyTerminal`, it use `Some((b"Body", id, part)`

```suggestion
				Some((b"Treasury", b"Voice").encode()),
```